### PR TITLE
API subpkg

### PR DESCRIPTION
**Pull request by Tal-or**

### CODE REVIEW

File: pkg/deployer/api/api.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
+package api
```
Comment (fromanirh) [inline pkg/deployer/api/api.go:1]: please add the copyright boilerplate

File: pkg/deployer/api/api.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+func Deploy(logger *log.Logger, opts *Options) error {
+	var err error
+
+	mf, err := GetManifest()
+	if err != nil {
+		return err
+	}
+	logger.Printf("  RTE manifests loaded")
+
+	hp, err := deployer.NewHelper("RTE")
```
Comment (fromanirh) [inline pkg/deployer/api/api.go]: please use "API" here, not "RTE":
```
hp, err := deployer.NewHelper("API")
```

File: pkg/deployer/api/api.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+func Remove(logger *log.Logger, opts *Options) error {
+	var err error
+
+	mf, err := GetManifest()
+	if err != nil {
+		return err
+	}
+	logger.Printf("  RTE manifests loaded")
+
+	hp, err := deployer.NewHelper("RTE")
```
Comment (fromanirh) [inline pkg/deployer/api/api.go]: same

File: pkg/commands/deploy.go [hunk 4]
```diff
@@ -17,13 +17,10 @@
 package commands
 
 import (
-	"context"
+	"github.com/fromanirh/deployer/pkg/deployer/api"
```
Comment (fromanirh) [inline pkg/commands/deploy.go:21]: remove blank line please, it should read like
```
"github.com/fromanirh/deployer/pkg/deployer/api"
"github.com/fromanirh/deployer/pkg/deployer/rte"
"github.com/fromanirh/deployer/pkg/deployer/sched"

"github.com/spf13/cobra"
```

File: pkg/commands/deploy.go [hunk 5]
```diff
@@ -48,10 +48,10 @@ func NewRemoveCommand(commonOpts *CommonOptions) *cobra.Command {
 		Use:   "remove",
 		Short: "remove the components and configurations needed for topology-aware-scheduling",
 		RunE: func(cmd *cobra.Command, args []string) error {
```
Comment (fromanirh) [inline pkg/commands/deploy.go:50]: in this case copying the option is intended, it emphasizes the fact the function will NOT mutate the argument.
Furthermore, in this case the copies are not really performance sensitive, so overall the added bit of clarity justifies the extra copy.

File: pkg/deployer/api/api.go [hunk 6]
```diff
@@ -39,9 +39,9 @@ func Deploy(logger *log.Logger, opts *Options) error {
 	if err != nil {
 		return err
 	}
-	logger.Printf("  RTE manifests loaded")
+	logger.Printf("  API manifests loaded")
```
Comment (fromanirh) [inline pkg/deployer/api/api.go:42]: good, but please squash in the initial commit

File: pkg/commands/deploy.go [hunk 7]
```diff
@@ -156,4 +147,17 @@ func deployOnCluster(commonOpts *CommonOptions, opts *deployOptions) error {
 		return err
 	}
 	return nil
+}
+
+func removeFromCluster(commonOpts *CommonOptions, opts *deployOptions) error {
```
Comment (fromanirh) [inline pkg/commands/deploy.go:152]: but this is used in just one place - `deployOnCluster` is used in two places, so is it really needed?

File: pkg/commands/deploy.go [hunk 8]
```diff
@@ -160,4 +160,4 @@ func removeFromCluster(commonOpts *CommonOptions, opts *deployOptions) error {
 		return err
 	}
 	return nil
-}
\ No newline at end of file
+}
```
Comment (fromanirh) [inline pkg/commands/deploy.go]: good, but plase squash in the previous commit (if we decide to keep it at all)

**Comment by fromanirh** [comment]
nice work, just minor comments but quite some of them

File: pkg/commands/deploy.go (hunk 5 above)
Comment (Tal-or) [reply pkg/commands/deploy.go:50]: Thank you for the clarification

**Review by fromanirh — COMMENTED** [review]
/approve
/lgtm
thanks for the updates

**Pull request merged**